Bug/fix correlation export

zsloan opened the pull request:
#### Description
This PR is intended to fix an issue where using Scroller caused the Correlation table export to not work.

While fixing this issue, I noticed a number of other issues related to search_results.js having a number of JQuery functions that needed to be changed to use the DataTables API. This PR should fix most of those, but there are a couple more I still need to address.

The one remaining issue I'm aware of is that Select All doesn't select all rows for tables using pagination or Scroller that have a number of results exceeding what appears on the first page (or first N results for Scroller). I'll fix this tomorrow morning.

#### How should this be tested?
Do correlations from the following link (or any other trait page) to test the correlation table and check that functions like Select All, Deselect, and clicking rows function as expected - https://genenetwork.org/show_trait?trait_id=10015&dataset=BXDPublish

Or do any search or view a collection totest those tables.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A

#### Questions

<!-- Are there any questions for the reviewer -->

zsloan committed:
Increased correlation table vertical height to 100vh

zsloan committed:
Convert trait/dataset names to strings, since phenotype names were being treated as integers

zsloan committed:
Converted the submit_traits_to_export_or_bnw function to using DataTables API, since the JQuery it was previously using didn't work with Scroller

zsloan committed:
Fixed select_all and deselect_all to use DataTables API so they work with any tables that use Scroller

zsloan committed:
Changed invert function to use DataTables API so it can also work with Scroller

zsloan committed:
Added Invert button to search result page

zsloan committed:
Fixed View Collection table drawCallback to correctly deal with check boxes and the Copy/Delete buttons and fixed an issue where extra white-space on either side of the checkboxes was causing some JS to not work correctly

zsloan committed:
Fixed drawCallback JS to work correctly and removed change_buttons function since it's now in search_results.js

zsloan committed:
Fixed drawCallback for correlation page table and added the Invert button

zsloan committed:
Changed a lot of the logic related to highlighting rows and disabling/enabling the Add/Copy/Remove buttons to use DataTables API

Moved change_buttons where it can be used outside of search_results.js

robwwilliams commented:
Great!

On Thu, Dec 3, 2020 at 5:18 PM zsloan <notifications@github.com> wrote:

> Description
>
> This PR is intended to fix an issue where using Scroller caused the
> Correlation table export to not work.
>
> While fixing this issue, I noticed a number of other issues related to
> search_results.js having a number of JQuery functions that needed to be
> changed to use the DataTables API. This PR should fix most of those, but
> there are a couple more I still need to address.
>
> The one remaining issue I'm aware of is that Select All doesn't select all
> rows for tables using pagination or Scroller that have a number of results
> exceeding what appears on the first page (or first N results for Scroller).
> I'll fix this tomorrow morning.
> How should this be tested?
>
> Do correlations from the following link (or any other trait page) to test
> the correlation table and check that functions like Select All, Deselect,
> and clicking rows function as expected -
> https://genenetwork.org/show_trait?trait_id=10015&dataset=BXDPublish
>
> Or do any search or view a collection totest those tables.
> Any background context you want to provide?
>
> N/A
> What are the relevant pivotal tracker stories?
>
> N/A
> Screenshots (if appropriate)
>
> N/A
> Questions
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/genenetwork/genenetwork2/pull/500
> Commit Summary
>
>    - Increased correlation table vertical height to 100vh
>    - Convert trait/dataset names to strings, since phenotype names were
>    being treated as integers
>    - Converted the submit_traits_to_export_or_bnw function to using
>    DataTables API, since the JQuery it was previously using didn't work with
>    Scroller
>    - Fixed select_all and deselect_all to use DataTables API so they work
>    with any tables that use Scroller
>    - Changed invert function to use DataTables API so it can also work
>    with Scroller
>    - Added Invert button to search result page
>    - Fixed View Collection table drawCallback to correctly deal with
>    check boxes and the Copy/Delete buttons and fixed an issue where extra
>    white-space on either side of the checkboxes was causing some JS to not
>    work correctly
>    - Fixed drawCallback JS to work correctly and removed change_buttons
>    function since it's now in search_results.js
>    - Fixed drawCallback for correlation page table and added the Invert
>    button
>    - Changed a lot of the logic related to highlighting rows and
>    disabling/enabling the Add/Copy/Remove buttons to use DataTables API
>
> File Changes
>
>    - *M* wqflask/wqflask/correlation/show_corr_results.py
>    <https://github.com/genenetwork/genenetwork2/pull/500/files#diff-3fdb436383cbd7c94e40080d2b8e1cf407332baf94349576ade6306ea97f129c>
>    (1)
>    - *M* wqflask/wqflask/export_traits.py
>    <https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8bf3b3f28df23ea0dcdd5a85fd2fdee487c87a3ec97535ba1f68bbedc75607e6>
>    (2)
>    - *M* wqflask/wqflask/static/new/javascript/search_results.js
>    <https://github.com/genenetwork/genenetwork2/pull/500/files#diff-041f9eb9c70a7a389ba15a37c10e07c75ad280343dff6d6bfa9afdf8376d15ff>
>    (165)
>    - *M* wqflask/wqflask/templates/collections/view.html
>    <https://github.com/genenetwork/genenetwork2/pull/500/files#diff-aa4b8fefc29ca60e160d68a51dbb662c14a47bfd0fa5de6d59b9795f41203923>
>    (33)
>    - *M* wqflask/wqflask/templates/correlation_page.html
>    <https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8e43a47f4db19817fb7ad62665dc2bf01330902bc58cb125b722da33647b02a0>
>    (54)
>    - *M* wqflask/wqflask/templates/search_result_page.html
>    <https://github.com/genenetwork/genenetwork2/pull/500/files#diff-4b981a44ca0c4740bf4187147b718c68ca4e1b4aba1434c72fccc36dbd5e2543>
>    (46)
>
> Patch Links:
>
>    - https://github.com/genenetwork/genenetwork2/pull/500.patch
>    - https://github.com/genenetwork/genenetwork2/pull/500.diff
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/500>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7ETRADYPRCPGEDHUFLSTAMDJANCNFSM4UMVM66Q>
> .
>
-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu  or  labwilliams@gmail.com

robwwilliams commented:
Hi,
Unfortunately, it's not quite fixed yet. 😭 The "download CSV" function downloads the data (i.e., the actual traits; see BXD_traits.csv as example), not the correlation results.

DETAILED DESCRIPTION OF ISSUE
The scroller is NOT the cause for this results export issue . This issue pre-existed the scroller update. What the scroller did was, it exacerbated an issue and has now made the correlation results pretty inaccessible for closer inspection. This is why:
1) Users can no longer copy the full results table and paste it to an excel sheet where they can scrutinize results and make tables for manuscripts, etc. Before scroller, I could select the full results table (like 2000 rows, or however many rows I want) and copy and paste to an excel sheet (see example file: exampleCorResults.xlsx). Now I can only copy parts of the table... that means, to get 2000 rows, I'll have to repeat copy-paste bit by bit as it only allows copying the part that's visible in the scroll window (like 70 rows on a small laptop).
2) Now the sort functions do not work. I thought a work-around would be to simply sort by chromosomal location and copy and paste from the region I'm interested in as a temporary solution, but the table is no longer sortable.

IDEAL SOLUTIONS
Ideally, when a user clicks "download csv", we want the actual table of correlation results that we're seeing, not the traits. So maybe an "export results" button would be great. Non-ideal solution would be to revert back to the pre-scroll format because at least then, the correlation results were fully accessible.

Being able to download the traits data is also useful (exactly what the download function currently does )... so maybe there could be a distinction. One option could be to label the current "download csv" as "download traits". Alternatively, "download traits" could be implemented when the traits are added to "collection".

Also, be great if the sort functions are re-implemented.

Beni


[https://uthsc.edu/brand/images/email-signature/shortsig-green-horizontal.png]

Khyobeni Mozhui, PhD
Assistant Professor

The University of Tennessee Health Science Center
Preventive Medicine
66 N. Pauline Street, #631
Memphis, TN 38163

kmozhui@uthsc.edu
t: 901.448.7334

________________________________
From: Rob Williams <rwilliams@uthsc.edu>
Sent: Thursday, December 3, 2020 8:08 PM
To: kmozhui <kmozhui@uthsc.edu>; genenetwork/genenetwork2 <reply@reply.github.com>
Subject: [Ext] Re: [genenetwork/genenetwork2] Bug/fix correlation export (#500)

Great!

On Thu, Dec 3, 2020 at 5:18 PM zsloan <notifications@github.com<mailto:notifications@github.com>> wrote:
Description

This PR is intended to fix an issue where using Scroller caused the Correlation table export to not work.

While fixing this issue, I noticed a number of other issues related to search_results.js having a number of JQuery functions that needed to be changed to use the DataTables API. This PR should fix most of those, but there are a couple more I still need to address.

The one remaining issue I'm aware of is that Select All doesn't select all rows for tables using pagination or Scroller that have a number of results exceeding what appears on the first page (or first N results for Scroller). I'll fix this tomorrow morning.

How should this be tested?

Do correlations from the following link (or any other trait page) to test the correlation table and check that functions like Select All, Deselect, and clicking rows function as expected - https://genenetwork.org/show_trait?trait_id=10015&dataset=BXDPublish

Or do any search or view a collection totest those tables.

Any background context you want to provide?

N/A

What are the relevant pivotal tracker stories?

N/A

Screenshots (if appropriate)

N/A

Questions
________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/genenetwork/genenetwork2/pull/500

Commit Summary

  *   Increased correlation table vertical height to 100vh
  *   Convert trait/dataset names to strings, since phenotype names were being treated as integers
  *   Converted the submit_traits_to_export_or_bnw function to using DataTables API, since the JQuery it was previously using didn't work with Scroller
  *   Fixed select_all and deselect_all to use DataTables API so they work with any tables that use Scroller
  *   Changed invert function to use DataTables API so it can also work with Scroller
  *   Added Invert button to search result page
  *   Fixed View Collection table drawCallback to correctly deal with check boxes and the Copy/Delete buttons and fixed an issue where extra white-space on either side of the checkboxes was causing some JS to not work correctly
  *   Fixed drawCallback JS to work correctly and removed change_buttons function since it's now in search_results.js
  *   Fixed drawCallback for correlation page table and added the Invert button
  *   Changed a lot of the logic related to highlighting rows and disabling/enabling the Add/Copy/Remove buttons to use DataTables API

File Changes

  *   M wqflask/wqflask/correlation/show_corr_results.py<https://github.com/genenetwork/genenetwork2/pull/500/files#diff-3fdb436383cbd7c94e40080d2b8e1cf407332baf94349576ade6306ea97f129c> (1)
  *   M wqflask/wqflask/export_traits.py<https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8bf3b3f28df23ea0dcdd5a85fd2fdee487c87a3ec97535ba1f68bbedc75607e6> (2)
  *   M wqflask/wqflask/static/new/javascript/search_results.js<https://github.com/genenetwork/genenetwork2/pull/500/files#diff-041f9eb9c70a7a389ba15a37c10e07c75ad280343dff6d6bfa9afdf8376d15ff> (165)
  *   M wqflask/wqflask/templates/collections/view.html<https://github.com/genenetwork/genenetwork2/pull/500/files#diff-aa4b8fefc29ca60e160d68a51dbb662c14a47bfd0fa5de6d59b9795f41203923> (33)
  *   M wqflask/wqflask/templates/correlation_page.html<https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8e43a47f4db19817fb7ad62665dc2bf01330902bc58cb125b722da33647b02a0> (54)
  *   M wqflask/wqflask/templates/search_result_page.html<https://github.com/genenetwork/genenetwork2/pull/500/files#diff-4b981a44ca0c4740bf4187147b718c68ca4e1b4aba1434c72fccc36dbd5e2543> (46)

Patch Links:

  *   https://github.com/genenetwork/genenetwork2/pull/500.patch
  *   https://github.com/genenetwork/genenetwork2/pull/500.diff

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/genenetwork/genenetwork2/pull/500>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AC55V7ETRADYPRCPGEDHUFLSTAMDJANCNFSM4UMVM66Q>.

--
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu<mailto:rwilliams@uthsc.edu>  or  labwilliams@gmail.com<mailto:labwilliams@gmail.com>

zsloan commented:
Ahh, I see. Scroller was causing the export to not work at all, so
that initially seemed to be the issue.

I just changed it to export the table contents. The Download button is now
on a separate line because it's strangely difficult to get it to appear on
the same line as the other buttons, but I wanted to go ahead and make the
change available sooner rather than later.  This export also currently only
contains table contents; I'll have to look up how to add metadata to it.

The sort seems to work correctly for me; what trait are you using?


On Fri, Dec 4, 2020 at 10:28 AM robwwilliams <notifications@github.com>
wrote:

> Hi,
> Unfortunately, it's not quite fixed yet. 😭 The "download CSV" function
> downloads the data (i.e., the actual traits; see BXD_traits.csv as
> example), not the correlation results.
>
> DETAILED DESCRIPTION OF ISSUE
> The scroller is NOT the cause for this results export issue . This issue
> pre-existed the scroller update. What the scroller did was, it exacerbated
> an issue and has now made the correlation results pretty inaccessible for
> closer inspection. This is why:
> 1) Users can no longer copy the full results table and paste it to an
> excel sheet where they can scrutinize results and make tables for
> manuscripts, etc. Before scroller, I could select the full results table
> (like 2000 rows, or however many rows I want) and copy and paste to an
> excel sheet (see example file: exampleCorResults.xlsx). Now I can only copy
> parts of the table... that means, to get 2000 rows, I'll have to repeat
> copy-paste bit by bit as it only allows copying the part that's visible in
> the scroll window (like 70 rows on a small laptop).
> 2) Now the sort functions do not work. I thought a work-around would be to
> simply sort by chromosomal location and copy and paste from the region I'm
> interested in as a temporary solution, but the table is no longer sortable.
>
> IDEAL SOLUTIONS
> Ideally, when a user clicks "download csv", we want the actual table of
> correlation results that we're seeing, not the traits. So maybe an "export
> results" button would be great. Non-ideal solution would be to revert back
> to the pre-scroll format because at least then, the correlation results
> were fully accessible.
>
> Being able to download the traits data is also useful (exactly what the
> download function currently does )... so maybe there could be a
> distinction. One option could be to label the current "download csv" as
> "download traits". Alternatively, "download traits" could be implemented
> when the traits are added to "collection".
>
> Also, be great if the sort functions are re-implemented.
>
> Beni
>
>
> [
> https://uthsc.edu/brand/images/email-signature/shortsig-green-horizontal.png]
>
>
> Khyobeni Mozhui, PhD
> Assistant Professor
>
> The University of Tennessee Health Science Center
> Preventive Medicine
> 66 N. Pauline Street, #631
> Memphis, TN 38163
>
> kmozhui@uthsc.edu
> t: 901.448.7334
>
> ________________________________
> From: Rob Williams <rwilliams@uthsc.edu>
> Sent: Thursday, December 3, 2020 8:08 PM
> To: kmozhui <kmozhui@uthsc.edu>; genenetwork/genenetwork2 <
> reply@reply.github.com>
> Subject: [Ext] Re: [genenetwork/genenetwork2] Bug/fix correlation export
> (#500)
>
> Great!
>
> On Thu, Dec 3, 2020 at 5:18 PM zsloan <notifications@github.com<mailto:
> notifications@github.com>> wrote:
> Description
>
> This PR is intended to fix an issue where using Scroller caused the
> Correlation table export to not work.
>
> While fixing this issue, I noticed a number of other issues related to
> search_results.js having a number of JQuery functions that needed to be
> changed to use the DataTables API. This PR should fix most of those, but
> there are a couple more I still need to address.
>
> The one remaining issue I'm aware of is that Select All doesn't select all
> rows for tables using pagination or Scroller that have a number of results
> exceeding what appears on the first page (or first N results for Scroller).
> I'll fix this tomorrow morning.
>
> How should this be tested?
>
> Do correlations from the following link (or any other trait page) to test
> the correlation table and check that functions like Select All, Deselect,
> and clicking rows function as expected -
> https://genenetwork.org/show_trait?trait_id=10015&dataset=BXDPublish
>
> Or do any search or view a collection totest those tables.
>
> Any background context you want to provide?
>
> N/A
>
> What are the relevant pivotal tracker stories?
>
> N/A
>
> Screenshots (if appropriate)
>
> N/A
>
> Questions
> ________________________________
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/genenetwork/genenetwork2/pull/500
>
> Commit Summary
>
> * Increased correlation table vertical height to 100vh
> * Convert trait/dataset names to strings, since phenotype names were being
> treated as integers
> * Converted the submit_traits_to_export_or_bnw function to using
> DataTables API, since the JQuery it was previously using didn't work with
> Scroller
> * Fixed select_all and deselect_all to use DataTables API so they work
> with any tables that use Scroller
> * Changed invert function to use DataTables API so it can also work with
> Scroller
> * Added Invert button to search result page
> * Fixed View Collection table drawCallback to correctly deal with check
> boxes and the Copy/Delete buttons and fixed an issue where extra
> white-space on either side of the checkboxes was causing some JS to not
> work correctly
> * Fixed drawCallback JS to work correctly and removed change_buttons
> function since it's now in search_results.js
> * Fixed drawCallback for correlation page table and added the Invert
> button
> * Changed a lot of the logic related to highlighting rows and
> disabling/enabling the Add/Copy/Remove buttons to use DataTables API
>
> File Changes
>
> * M wqflask/wqflask/correlation/show_corr_results.py<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-3fdb436383cbd7c94e40080d2b8e1cf407332baf94349576ade6306ea97f129c>
> (1)
> * M wqflask/wqflask/export_traits.py<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8bf3b3f28df23ea0dcdd5a85fd2fdee487c87a3ec97535ba1f68bbedc75607e6>
> (2)
> * M wqflask/wqflask/static/new/javascript/search_results.js<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-041f9eb9c70a7a389ba15a37c10e07c75ad280343dff6d6bfa9afdf8376d15ff>
> (165)
> * M wqflask/wqflask/templates/collections/view.html<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-aa4b8fefc29ca60e160d68a51dbb662c14a47bfd0fa5de6d59b9795f41203923>
> (33)
> * M wqflask/wqflask/templates/correlation_page.html<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8e43a47f4db19817fb7ad62665dc2bf01330902bc58cb125b722da33647b02a0>
> (54)
> * M wqflask/wqflask/templates/search_result_page.html<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-4b981a44ca0c4740bf4187147b718c68ca4e1b4aba1434c72fccc36dbd5e2543>
> (46)
>
> Patch Links:
>
> * https://github.com/genenetwork/genenetwork2/pull/500.patch
> * https://github.com/genenetwork/genenetwork2/pull/500.diff
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub<
> https://github.com/genenetwork/genenetwork2/pull/500>, or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/AC55V7ETRADYPRCPGEDHUFLSTAMDJANCNFSM4UMVM66Q>.
>
>
> --
> Rob
>
> Robert W. Williams, Ph.D.
> Chair: Department of Genetics, Genomics and Informatics
> University of Tennessee Health Science Center
> 71 S Manassas St, Memphis TN 38103-3310, TSRB Room 405
> Tele/Cell: 901 604 4752
> EMAIL: rwilliams@uthsc.edu<mailto:rwilliams@uthsc.edu> or
> labwilliams@gmail.com<mailto:labwilliams@gmail.com>
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/500#issuecomment-738878909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGHQLYRHTOKAKBXFYJ3STEE2BANCNFSM4UMVM66Q>
> .
>

robwwilliams commented:
Hi,
Just tested. Works great now. The "download" now downloads the results. And the sort buttons are working
Great job!

Thanks

Beni


[https://uthsc.edu/brand/images/email-signature/shortsig-green-horizontal.png]

Khyobeni Mozhui, PhD
Assistant Professor

The University of Tennessee Health Science Center
Preventive Medicine
66 N. Pauline Street, #631
Memphis, TN 38163

kmozhui@uthsc.edu
t: 901.448.7334

________________________________
From: Rob Williams <rwilliams@uthsc.edu>
Sent: Friday, December 4, 2020 2:11 PM
To: genenetwork/genenetwork2 <reply@reply.github.com>; kmozhui <kmozhui@uthsc.edu>
Cc: genenetwork/genenetwork2 <genenetwork2@noreply.github.com>; Review requested <review_requested@noreply.github.com>
Subject: [Ext] Re: [genenetwork/genenetwork2] Bug/fix correlation export (#500)



On Fri, Dec 4, 2020 at 2:01 PM zsloan <notifications@github.com<mailto:notifications@github.com>> wrote:

Ahh, I see. Scroller was causing the export to not work at all, so
that initially seemed to be the issue.

I just changed it to export the table contents. The Download button is now
on a separate line because it's strangely difficult to get it to appear on
the same line as the other buttons, but I wanted to go ahead and make the
change available sooner rather than later. This export also currently only
contains table contents; I'll have to look up how to add metadata to it.

The sort seems to work correctly for me; what trait are you using?


On Fri, Dec 4, 2020 at 10:28 AM robwwilliams <notifications@github.com<mailto:notifications@github.com>>
wrote:

> Hi,
> Unfortunately, it's not quite fixed yet. 😭 The "download CSV" function
> downloads the data (i.e., the actual traits; see BXD_traits.csv as
> example), not the correlation results.
>
> DETAILED DESCRIPTION OF ISSUE
> The scroller is NOT the cause for this results export issue . This issue
> pre-existed the scroller update. What the scroller did was, it exacerbated
> an issue and has now made the correlation results pretty inaccessible for
> closer inspection. This is why:
> 1) Users can no longer copy the full results table and paste it to an
> excel sheet where they can scrutinize results and make tables for
> manuscripts, etc. Before scroller, I could select the full results table
> (like 2000 rows, or however many rows I want) and copy and paste to an
> excel sheet (see example file: exampleCorResults.xlsx). Now I can only copy
> parts of the table... that means, to get 2000 rows, I'll have to repeat
> copy-paste bit by bit as it only allows copying the part that's visible in
> the scroll window (like 70 rows on a small laptop).
> 2) Now the sort functions do not work. I thought a work-around would be to
> simply sort by chromosomal location and copy and paste from the region I'm
> interested in as a temporary solution, but the table is no longer sortable.
>
> IDEAL SOLUTIONS
> Ideally, when a user clicks "download csv", we want the actual table of
> correlation results that we're seeing, not the traits. So maybe an "export
> results" button would be great. Non-ideal solution would be to revert back
> to the pre-scroll format because at least then, the correlation results
> were fully accessible.
>
> Being able to download the traits data is also useful (exactly what the
> download function currently does )... so maybe there could be a
> distinction. One option could be to label the current "download csv" as
> "download traits". Alternatively, "download traits" could be implemented
> when the traits are added to "collection".
>
> Also, be great if the sort functions are re-implemented.
>
> Beni
>
>
> [
> https://uthsc.edu/brand/images/email-signature/shortsig-green-horizontal.png]
>
>
> Khyobeni Mozhui, PhD
> Assistant Professor
>
> The University of Tennessee Health Science Center
> Preventive Medicine
> 66 N. Pauline Street, #631
> Memphis, TN 38163
>
> kmozhui@uthsc.edu<mailto:kmozhui@uthsc.edu>
> t: 901.448.7334
>
> ________________________________
> From: Rob Williams <rwilliams@uthsc.edu<mailto:rwilliams@uthsc.edu>>
> Sent: Thursday, December 3, 2020 8:08 PM
> To: kmozhui <kmozhui@uthsc.edu<mailto:kmozhui@uthsc.edu>>; genenetwork/genenetwork2 <
> reply@reply.github.com<mailto:reply@reply.github.com>>
> Subject: [Ext] Re: [genenetwork/genenetwork2] Bug/fix correlation export
> (#500)
>
> Great!
>
> On Thu, Dec 3, 2020 at 5:18 PM zsloan <notifications@github.com<mailto:notifications@github.com><mailto:
> notifications@github.com<mailto:notifications@github.com>>> wrote:
> Description
>
> This PR is intended to fix an issue where using Scroller caused the
> Correlation table export to not work.
>
> While fixing this issue, I noticed a number of other issues related to
> search_results.js having a number of JQuery functions that needed to be
> changed to use the DataTables API. This PR should fix most of those, but
> there are a couple more I still need to address.
>
> The one remaining issue I'm aware of is that Select All doesn't select all
> rows for tables using pagination or Scroller that have a number of results
> exceeding what appears on the first page (or first N results for Scroller).
> I'll fix this tomorrow morning.
>
> How should this be tested?
>
> Do correlations from the following link (or any other trait page) to test
> the correlation table and check that functions like Select All, Deselect,
> and clicking rows function as expected -
> https://genenetwork.org/show_trait?trait_id=10015&dataset=BXDPublish
>
> Or do any search or view a collection totest those tables.
>
> Any background context you want to provide?
>
> N/A
>
> What are the relevant pivotal tracker stories?
>
> N/A
>
> Screenshots (if appropriate)
>
> N/A
>
> Questions
> ________________________________
> You can view, comment on, or merge this pull request online at:
>
> https://github.com/genenetwork/genenetwork2/pull/500
>
> Commit Summary
>
> * Increased correlation table vertical height to 100vh
> * Convert trait/dataset names to strings, since phenotype names were being
> treated as integers
> * Converted the submit_traits_to_export_or_bnw function to using
> DataTables API, since the JQuery it was previously using didn't work with
> Scroller
> * Fixed select_all and deselect_all to use DataTables API so they work
> with any tables that use Scroller
> * Changed invert function to use DataTables API so it can also work with
> Scroller
> * Added Invert button to search result page
> * Fixed View Collection table drawCallback to correctly deal with check
> boxes and the Copy/Delete buttons and fixed an issue where extra
> white-space on either side of the checkboxes was causing some JS to not
> work correctly
> * Fixed drawCallback JS to work correctly and removed change_buttons
> function since it's now in search_results.js
> * Fixed drawCallback for correlation page table and added the Invert
> button
> * Changed a lot of the logic related to highlighting rows and
> disabling/enabling the Add/Copy/Remove buttons to use DataTables API
>
> File Changes
>
> * M wqflask/wqflask/correlation/show_corr_results.py<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-3fdb436383cbd7c94e40080d2b8e1cf407332baf94349576ade6306ea97f129c>
> (1)
> * M wqflask/wqflask/export_traits.py<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8bf3b3f28df23ea0dcdd5a85fd2fdee487c87a3ec97535ba1f68bbedc75607e6>
> (2)
> * M wqflask/wqflask/static/new/javascript/search_results.js<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-041f9eb9c70a7a389ba15a37c10e07c75ad280343dff6d6bfa9afdf8376d15ff>
> (165)
> * M wqflask/wqflask/templates/collections/view.html<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-aa4b8fefc29ca60e160d68a51dbb662c14a47bfd0fa5de6d59b9795f41203923>
> (33)
> * M wqflask/wqflask/templates/correlation_page.html<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-8e43a47f4db19817fb7ad62665dc2bf01330902bc58cb125b722da33647b02a0>
> (54)
> * M wqflask/wqflask/templates/search_result_page.html<
> https://github.com/genenetwork/genenetwork2/pull/500/files#diff-4b981a44ca0c4740bf4187147b718c68ca4e1b4aba1434c72fccc36dbd5e2543>
> (46)
>
> Patch Links:
>
> * https://github.com/genenetwork/genenetwork2/pull/500.patch
> * https://github.com/genenetwork/genenetwork2/pull/500.diff
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub<
> https://github.com/genenetwork/genenetwork2/pull/500>, or unsubscribe<
> https://github.com/notifications/unsubscribe-auth/AC55V7ETRADYPRCPGEDHUFLSTAMDJANCNFSM4UMVM66Q>.
>
>
> --
> Rob
>
> Robert W. Williams, Ph.D.
> Chair: Department of Genetics, Genomics and Informatics
> University of Tennessee Health Science Center
> 71 S Manassas St, Memphis TN 38103-3310, TSRB Room 405
> Tele/Cell: 901 604 4752
> EMAIL: rwilliams@uthsc.edu<mailto:rwilliams@uthsc.edu><mailto:rwilliams@uthsc.edu<mailto:rwilliams@uthsc.edu>> or
> labwilliams@gmail.com<mailto:labwilliams@gmail.com><mailto:labwilliams@gmail.com<mailto:labwilliams@gmail.com>>
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork2/pull/500#issuecomment-738878909>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGHQLYRHTOKAKBXFYJ3STEE2BANCNFSM4UMVM66Q>
> .
>

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/genenetwork/genenetwork2/pull/500#issuecomment-738989349>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AC55V7BUMGVN5AJCZXSCI4TSTE5ZZANCNFSM4UMVM66Q>.


--
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: rwilliams@uthsc.edu<mailto:rwilliams@uthsc.edu>  or  labwilliams@gmail.com<mailto:labwilliams@gmail.com>
